### PR TITLE
Avoid self-referential imports.

### PR DESCRIPTION
**Pull request by Hixie**
And add a test to verify we don't do this again.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @tvolkert

File: dev/bots/test.dart [hunk 1]
```diff
@@ -241,6 +243,132 @@ Future<Null> _runFlutterAnalyze(String workingDirectory, {
...
+      print('${bold}Multiple errors were detected when looking at import dependencies within the Flutter package:$reset\n');
+    }
+    print(errors.join('\n\n'));
+    print('$red━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━$reset\n');
+    exit(1);
+  }
+}
+
+bool _matches<T>(List<T> a, List<T> b) {
+  assert(a != null);
```
Comment (tvolkert) [inline dev/bots/test.dart]:
```
a.length == b.length && a.asMap().keys.every((int i) => a[i] == b[i])
```

File: dev/bots/test.dart [hunk 2]
```diff
@@ -241,6 +243,132 @@ Future<Null> _runFlutterAnalyze(String workingDirectory, {
...
+  return true;
+}
+
+final RegExp _importPattern = new RegExp(r"import 'package:flutter/([^.]+)\.dart'");
+final RegExp _importMetaPattern = new RegExp(r"import 'package:meta/meta.dart'");
+
+Set<String> _findDependencies(String srcPath, List<String> errors, { bool checkForMeta: false }) {
+  return new Directory(srcPath).listSync().where((FileSystemEntity entity) {
+    return entity is File && path.extension(entity.path) == '.dart';
+  }).map<Set<String>>((FileSystemEntity entity) {
```
Comment (tvolkert) [inline dev/bots/test.dart]: I think you can just make this take `File file` since you have the `entity is File` check above

**Review by tvolkert — APPROVED** [review]
LGTM

File: dev/bots/test.dart (hunk 1 above)
Comment (Hixie) [inline dev/bots/test.dart]: How strongly do you feel about this? I think the code I have now is both more efficient and more readable, but I don't feel strongly about this.

File: dev/bots/test.dart (hunk 2 above)
Comment (Hixie) [inline dev/bots/test.dart]: [error] The function expression type '(File) → Set<String>' isn't of type '(FileSystemEntity) → Set<String>'. This means its parameter or return type does not match what is expected. Consider changing parameter type(s) or the returned type(s).

File: dev/bots/test.dart (hunk 1 above)
Comment (tvolkert) [inline dev/bots/test.dart]: I like the brevity, but I don't feel strongly about it, so your call.

**Pull request merged**